### ISSUE TITLE
Allow producing intermediate ADTs instead of only HTML

### ISSUE THREAD

**Issue by hdgarrood** [paf31/psc-pages#12]
Something like this, perhaps?
```haskell
data RenderedTypeElement
  = Syntax Text
  | Ident Text (Maybe (Text, Text)) -- ^ Ident text and optional (module, declaration) to link to.
  | Keyword Text
  | Space
  
type RenderedType = [RenderedTypeElement]

asHTML :: RenderedType -> Html
asText :: RenderedType -> Text
```

The goal would be to allow rendering declarations as plain text, which makes testing Pursuit easier, and also enables generation of a plain text version of the docs, which can be used as input for a Hoogle database.

**Comment by paf31** [comment]
Sounds good.

**Comment by hdgarrood** [comment]
Fixed by #14
